### PR TITLE
add Ruby on Rails example

### PR DESCRIPTION
**Pull request by RajivSah**
https://github.com/earthly/earthly/issues/344

### CODE REVIEW

File: examples/ruby-on-rails/Earthfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    SAVE ARTIFACT . /.
+    SAVE IMAGE
+docker:
+    FROM +build
+    EXPOSE 3000
+    CMD ["bundle", "exec", "rails", "server", "-b", "0.0.0.0"]
+    SAVE IMAGE ruby-on-rails-example:latest
+migrate:
+    FROM +build
+    RUN bundle exec rails db:migrate
```
Comment (vladaionescu) [inline examples/ruby-on-rails/Earthfile:26]: This should probably be a `RUN --push` command perhaps?

File: examples/ruby-on-rails/Earthfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+test:
+    FROM +build
+    RUN bundle exec rails test
+console:
+    FROM +build
+    RUN bundle exec rails console
```
Comment (vladaionescu) [inline examples/ruby-on-rails/Earthfile:32]: Possibly here too - though we might be abusing the pattern.

`RUN --push` ensures that the command is not cached and will run it every time.

File: examples/ruby-on-rails/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+* Database initialization
+
+* How to run the test suite
+
+* Services (job queues, cache servers, search engines, etc.)
+
+* Deployment instructions
+
+* ...
```
Comment (vladaionescu) [inline examples/ruby-on-rails/README.md:24]: This README seems like boilerplate. Perhaps it's best if it's removed.

File: examples/ruby-on-rails/Earthfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+deps:
+    RUN curl -sL https://deb.nodesource.com/setup_12.x  | bash - && \
+      apt-get install -y nodejs
+    RUN gem install bundler:2.1.4
+    COPY Gemfile Gemfile.lock ./
+    RUN bundle config build.nokogiri --use-system-libraries
+    RUN bundle check || bundle install
+    SAVE ARTIFACT Gemfile.lock AS LOCAL ./Gemfile.lock
+    SAVE IMAGE
+build:
```
Comment (vladaionescu) [inline examples/ruby-on-rails/Earthfile:14]: Mind adding a blank line between targets? It helps keep the examples consistent in style.

**Review by vladaionescu — COMMENTED** [review]
Thanks for the contribution! 🙌 

I left a few comments. I'll leave the Ruby review to @agbell

File: examples/ruby-on-rails/Earthfile (hunk 4 above)
Comment (RajivSah) [reply examples/ruby-on-rails/Earthfile:15]: done

**Comment by agbell** [comment]
> I left a few comments. I'll leave the Ruby review to @agbell
@RajivSah Thanks for your contributions.  Can you address the requests @vladaionescu had around ``--push`` and blank lines?

**Review by agbell — CHANGES_REQUESTED** [review]
@RajivSah Thanks for your contributions.  Can you address the requests @vladaionescu had around ``--push`` and blank lines?

File: examples/ruby-on-rails/Earthfile (hunk 2 above)
Comment (vladaionescu) [reply examples/ruby-on-rails/Earthfile]: Actually this one is probably not a good idea since it's not interacting with an external service.

Comment (RajivSah) [reply examples/ruby-on-rails/Earthfile]: Do you mean to remove `--push`? @vladaionescu

Comment (vladaionescu) [reply examples/ruby-on-rails/Earthfile]: Yes - it should only be used for db:migrate, I think. But @agbell might know the Ruby patterns better.

**Comment by RajivSah** [comment]
> @RajivSah Thanks for your contributions. Can you address the requests @vladaionescu had around `--push` and blank lines?

@agbell Addressed

**Review by agbell — APPROVED** [review]

**Comment by agbell** [comment]
Apologies on the delay @RajivSah   Thank you for your contribution!

**Comment by agbell** [comment]
Hi @RajivSah Would you like us to send you an earthly sticker?  If so please send me your address to adam@earthly.dev

**Comment by RajivSah** [comment]
@agbell Of course! I would love it. 😊

**Pull request merged**